### ISSUE TITLE
Compact SeedQR sometimes fails

### ISSUE THREAD

**Issue by stepansnigirev** [cryptoadvance/specter-diy#211]
Maybe some magic bytes at start or at the end are breaking it.

**Comment by stepansnigirev** [comment]
Add more debug info:
![image](https://user-images.githubusercontent.com/1706012/180855766-d70f8dd5-de68-44c5-84f4-46b9b7267183.png)

Check this:
![image](https://user-images.githubusercontent.com/1706012/180855819-ba2e4409-168f-4ff5-b285-d04348586823.png)

Failing on this QR:
![image](https://user-images.githubusercontent.com/1706012/180855897-13723948-5285-457d-b330-440ec2fd42b7.png)

**Comment by kravens** [comment]
25x25 version of the same SeedQR has no issues importing, seems indeed only Compact QR related.
![image](https://user-images.githubusercontent.com/9646902/180856956-5b9388bb-69ef-4287-93b0-372f12f35fe1.png)
Here is the same 12-word seed as 25x25, which imports just fine.

https://github.com/SeedSigner/seedsigner/blob/dev/docs/seed_qr/README.md#recoverability
Since Compact SeedQR's use binary format instead of a human readable digit-stream, it is likely some strange unintended character that causes the bug.

**Comment by kdmukai** [comment]
Interesting. zxing.org was fine decoding it. Same for SeedSigner itself. Any clue which bytes were tripping up the DIY?

For testing on SeedSigner I wrote a script to randomly generate new seeds and their resulting Compact SeedQRs and fed those back into our decoder. That's how I uncovered the issue with `\x00` chars which required some tweaking of an updated version of `zbar` to fix (if I remember right). Once that char was handled, I ran the script out to maybe 100k random seeds and didn't have any further decoding problems so I figured the SeedSigner implementation was pretty safe at that point.

<img width="1013" alt="Screen Shot 2022-07-25 at 3 01 25 PM" src="https://user-images.githubusercontent.com/934746/180864381-467bebf2-2f43-4f6f-a72e-be6792d97a32.png">
